src/AppInstallerRepositoryCore/CompositeSource.cpp

```diff
@@ -436,92 +436,100 @@ namespace AppInstaller::Repository
-                                }
-                                else
+                                AICLI_LOG(Repo, Info, << "  Checking match with package id: " <<
+                                    availableMatch.Package->GetLatestAvailableVersion()->GetProperty(PackageVersionProperty::Id));
+
+                                if (IsStrongMatchField(availableMatch.MatchCriteria.Field))
                                 {
-                                    AICLI_LOG(Repo, Info, << "  Found multiple packages with strong match fields");
-                                    availablePackage.reset();
-                                    break;
+                                    if (!availablePackage)
+                                    {
+                                        availablePackage = std::move(availableMatch.Package);
+                                    }
+                                    else
+                                    {
+                                        AICLI_LOG(Repo, Info, << "  Found multiple packages with strong match fields");
+                                        availablePackage.reset();
+                                        break;
+                                    }
                                 }
                             }
-                        }
 
-                        if (!availablePackage)
-                        {
-                            AICLI_LOG(Repo, Warning, << "  Appropriate available package could not be determined");
+                            if (!availablePackage)
+                            {
+                                AICLI_LOG(Repo, Warning, << "  Appropriate available package could not be determined");
+                            }
                         }
+
+                        // We found some matching packages here, don't keep going
+                        break;
                     }
 
-                    // We found some matching packages here, don't keep going
-                    break;
+                    compositePackage->SetAvailablePackage(std::move(availablePackage));
                 }
 
-                compositePackage->SetAvailablePackage(std::move(availablePackage));
+                // Move the installed result into the composite result
+                result.Matches.emplace_back(std::move(compositePackage), std::move(match.MatchCriteria));
             }
 
-            // Move the installed result into the composite result
-            result.Matches.emplace_back(std::move(compositePackage), std::move(match.MatchCriteria));
-        }
-
-        // Optimization for the "everything installed" case, no need to allow for reverse correlations
-        if (request.IsForEverything())
-        {
-            return std::move(result);
+            // Optimization for the "everything installed" case, no need to allow for reverse correlations
+            if (request.IsForEverything())

```

Comment:
>if (request.IsForEverything()) [](http://example.com/codeflow?start=12&length=30)

This probably needs to not happen for `m_searchBehavior == CompositeSearchBehavior::AllPackages`, right?